design/baremetal-operator/retry-interface.md

```diff
@@ -0,0 +1,92 @@
+<!--
+ This work is licensed under a Creative Commons Attribution 3.0
+ Unported License.
+
+ http://creativecommons.org/licenses/by/3.0/legalcode
+-->
+
+# Title
+
+Retry from error state
+
+## Status
+
+provisional
+
+## Summary
+
+This proposal aims to introduce a new declarative API
+to forcibly retry the reconciliation of a waiting BareMetalHost resource
+(being in error state) when a fix is immediately available, so that it
+could be possible to reduce the waiting time required
```

Comment:
I'm still not clear why we need a special API for this. Any edit to the host, including adding a temporary label, would trigger reconciliation.

Reply:
I guess the difference this makes is that if it fails again for a different (perhaps more easily recoverable) reason then the time to retry is shorter.
That said, I must confess to being [somewhat sceptical](https://github.com/metal3-io/baremetal-operator/issues/798#issuecomment-806048253) of the value also.

Reply:
Yes, the key point of the proposal it's about resetting the ErrorCount to 1 - otherwise any edit will be fine fo retriggering the reconciliation.

Reply:
Why do we want to artificially reset the error count? Just to shorten retry intervals if there's another failure?

Reply:
Yes (or another set of failures)